### PR TITLE
Improve support for property highlighting

### PR DESCRIPTION
**Pull request by Alhadis**
__Adds:__  

*	Highlighting to `get`/`set` operators:

	```js
	set name(i){ }
	get name( ){ }
	```

*	Support for computed method names (ES6):

	```js
	const object = {
		    [ "na" + "me" ](){ }
		get [ "na" + "me" ](){ }
	};
	```

__Preview:__  
Before and after

__Question:__
How long until the scope-names of `[`, `{` and `(` can be fixed to use `punctuation` instead of the painfully-incorrect `meta.brace`...? It's inconsistent not just with other grammars, but also in how it affects highlighting between syntax themes.

I mean, this says it all:

Pls

### CODE REVIEW

**Comment by 50Wliu** [comment]
New scopes use `punctuation`, old scopes in patterns that are updated are generally updated to use `punctuation` as well.  So in this case you can change them to punctuation.

File: grammars/javascript.cson [hunk 1]
```diff
@@ -468,10 +468,25 @@
...
+          \\]
+        )
+        \\s*\\(\\s*               # Start of arguments list
+          (
+            "[^"]*"  |            # Double-quoted string
+            '[^']*'  |            # Single-quoted string
+             [^"()']              # Any non-bracket or non-quote
+          )*
+        \\)\\s*                   # End of arguments
+        \\{                       # Beginning of body
```
Comment (50Wliu) [inline grammars/javascript.cson]: {

**Comment by Alhadis** [comment]
Ace. Give me a sec.

File: grammars/javascript.cson [hunk 2]
```diff
@@ -488,6 +503,29 @@
         'name': 'meta.function.method.definition.js'
         'patterns': [
           {
+            'match': '(\\[)(.+)(\\])(?=\\s*\\()'
+            'captures':
+              '1':
+                'name': 'meta.brace.square.js'
```
Comment (50Wliu) [inline grammars/javascript.cson]: Use punctuation scopes

File: grammars/javascript.cson [hunk 3]
```diff
@@ -488,6 +503,29 @@
...
+                  {
+                    'include': '$self'
+                  }
+                  {
+                    'match': '[a-zA-Z_$][\\w$]*'
+                    'name': 'variable.parameter.property.js'
+                  }
+                ]
+              '3':
+                'name': 'meta.brace.square.js'
```
Comment (50Wliu) [inline grammars/javascript.cson]: Use punctuation scopes

File: grammars/javascript.cson (hunk 1 above)
Comment (Alhadis) [inline grammars/javascript.cson]: Heh, I always forget to unescape those for some reason...

**Comment by 50Wliu** [comment]
Can there also be a test with just a computed property name (without a getter/setter in front)?

**Comment by Alhadis** [comment]
Done. Really thankful I [wrote this](https://github.com/Alhadis/Atom-PhoenixTheme/commit/85f793feb8ed93cc4fb67b9110e8dd7266bb2bc6), BTW. :p

**Comment by 50Wliu** [comment]
> Done. Really thankful I wrote this, BTW. :p

To each their own :).  Just waiting for the build at this point.

**Pull request merged**